pkg/p2p/libp2p/libp2p.go

```diff
@@ -348,6 +347,25 @@ func buildUnderlayAddress(addr ma.Multiaddr, peerID libp2ppeer.ID) (ma.Multiaddr
 	return addr.Encapsulate(hostAddr), nil
 }
 
+func (s *Service) ConnectNotify(ctx context.Context, addr ma.Multiaddr) (address *bzz.Address, err error) {
```

Comment:
added a comment in the PR description, and also in the p2p interface that defines this method :+1:

Reply:
I had another suggestions for these two method calls, but do not have strong reasons against current ones. The only one is that ConnectNotify is more frequently used and can be shorter. Something like:

- `ConnectNotify` -> `Connect`
- `Connect` -> `ConnectSilently`

But is see why the current names can be more clear. I am just writing to see if there would be other opinions.

Reply:
I finally concluded on `Connect` and `ConnectNotify` because:

The default behavior should be that when we connect to another node we should not notify. The exception to this rule is the debug api and `node.go` which both need to connect and notify. The code in `node.go` that calls the connection to bootnodes can be eliminated in the future, by adding a `AddBootnode` method on topology (which can then selectively connect to bootnodes whenever needed. this in fact should not be up to the code in `node.go` to decide), leaving the only exception to the rule of connecting and not notifying to be in the debug api.

Reply:
it does not make sense to me that we could call `ConnectNotify` and not actually notify. the notification still seems to be dependant on an `if`. 

why?

Reply:
so @acud has explained this to me: the `topologyNotifier` could be `nil`, but it shouldn't be. the `if` is justified in order to prevent a `panic` (which should not happen, but in theory could)